Assets/__Scripts/MapEditor/AudioTimeSyncController.cs

```diff
@@ -347,27 +376,36 @@ public void TogglePlaying()
         IsPlaying = !IsPlaying;
         if (IsPlaying)
         {
-            if (CurrentSeconds >= SongAudioSource.clip.length - 0.1f)
+            //ELECAST TEST if (CurrentSeconds >= SongAudioSource.clip.length - 0.1f)
+            if (CurrentSeconds >= GetTotalSeconds() - 0.1f)//ELECAST TEST
             {
                 ResetTime();
             }
 
             playStartTime = CurrentSeconds;
-            SongAudioSource.time = CurrentSeconds;
-            SongAudioSource.Play();
+            //ELECAST TEST SongAudioSource.time = CurrentSeconds;
+            //ELECAST TEST SongAudioSource.Play();
+            SetVorbisPosition((long)(CurrentSeconds * bytesPerSecond));//ELECAST TEST
```

Comment:
seeking based on byte position instead of time might be inaccurate with variable bitrates
is there no way to seek based on time?

Reply:
variable bitrates is not a problem since bytesPerSecond is obtained when we load the song file, or do you mean something else? I couldn't find a way to seek based on time

Reply:
bitrate can vary throughout one file, not just between different files
by basing on the average bitrate, you can end up seeking to the wrong time

Reply:
Oh this makes sence, will check into that